subsys/nrf_security/src/psa_crypto_driver_wrappers.c

```diff
@@ -2436,7 +2436,9 @@ psa_status_t psa_driver_wrapper_get_random(psa_driver_random_context_t *context,
 					   size_t output_size)
 {
 #if defined(PSA_CRYPTO_DRIVER_ALG_PRNG_TEST)
-	return prng_test_generate_random(output, output_size);
+	if (prng_test_generate_random(output, output_size) == PSA_SUCCESS) {
+		return PSA_SUCCESS;
+	}
 #endif
 #if defined(PSA_CRYPTO_DRIVER_ALG_PRNG_CC3XX_PLATFORM)
 	int err;
```

Comment:
Usually fall-through is implemented with the driver returning PSA_ERROR_NOT_SUPPORTED and the wrapper checking for it, instead of checking ! SUCCESS.

Is it possible for the driver to return NOT_SUPPORTED so that we implement fall-through in a consistent way in the driver_wrappers?

Reply:
Yes, good point